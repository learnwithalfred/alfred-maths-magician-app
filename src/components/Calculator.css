body {
  background-color: whitesmoke;
}

.home-container {
  display: flex;
  justify-content: space-evenly;
  margin: 2rem;
  font-size: 2rem;
  flex-wrap: wrap;
}
