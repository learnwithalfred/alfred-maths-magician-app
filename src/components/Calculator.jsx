import React from 'react';
import InputElement from './InputElement';

class Calculator extends React.Component {
  state = {};
  render() {
    const rowStyles = {
      width: '100%',
      color: 'black',
      display: 'flex',
      flexWrap: 'nowrap',
    };

    const containerStyles = {
      width: '500px',
    };

    const orangeColor = '#F5913E';
    return (
      <>
        <div style={containerStyles}>
          <div style={rowStyles}>
            <InputElement
              width="100%"
              backgroundColor="#858694"
              value="0"
              elementPosition="flex-end"
            />
          </div>
          <div style={rowStyles}>
            <InputElement value="AC" />
            <InputElement value="+/-" />
            <InputElement value="%" />
            <InputElement value={'รท'} backgroundColor={orangeColor} />
          </div>

          <div style={rowStyles}>
            <InputElement value={7} />
            <InputElement value={8} />
            <InputElement value={9} />
            <InputElement value={'X'} backgroundColor={orangeColor} />
          </div>

          <div style={rowStyles}>
            <InputElement value={4} />
            <InputElement value={5} />
            <InputElement value={6} />
            <InputElement value={'-'} backgroundColor={orangeColor} />
          </div>

          <div style={rowStyles}>
            <InputElement value={1} />
            <InputElement value={2} />
            <InputElement value={3} />
            <InputElement value={'+'} backgroundColor={orangeColor} />
          </div>

          <div style={rowStyles}>
            <InputElement value={0} width="53%" />
            <InputElement value={'.'} />
            <InputElement value={'='} backgroundColor={orangeColor} />
          </div>
        </div>
      </>
    );
  }
}

export default Calculator;
