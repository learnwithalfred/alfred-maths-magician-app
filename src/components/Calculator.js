/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import InputElement from './InputElement';
import calculate from '../logic/calculate';
import './Calculator.css';
import Button from './button';

const Calculator = () => {
  const [obj, setObj] = React.useState({
    total: null,
    next: null,
    operation: null,
  });

  const handleClick = (buttonName) => {
    setObj((prevState) => {
      const results = {
        ...prevState,
        ...calculate(prevState, buttonName),
      };
      return results;
    });
  };

  const results = obj.next ? obj.next : obj.total;
  const rowStyles = {
    width: '100%',
    color: 'black',
    display: 'flex',
    flexWrap: 'nowrap',
  };

  const containerStyles = {
    width: '500px',
  };

  const orangeColor = '#F5913E';

  return (
    <>
      <div style={containerStyles}>
        <div style={rowStyles}>
          <InputElement
            width="100%"
            backgroundColor="#858694"
            value={results}
            elementPosition="flex-end"
          />
        </div>
        <div style={rowStyles}>
          <Button text="AC" handleClick={handleClick} />
          <Button handleClick={handleClick} text="+/-" />
          <Button handleClick={handleClick} text="%" />
          <Button
            handleClick={handleClick}
            text="รท"
            backgroundColor={orangeColor}
          />
        </div>

        <div style={rowStyles}>
          <Button handleClick={handleClick} text="7" />
          <Button handleClick={handleClick} text="8" />
          <Button handleClick={handleClick} text="9" />
          <Button
            handleClick={handleClick}
            text="x"
            backgroundColor={orangeColor}
          />
        </div>

        <div style={rowStyles}>
          <Button handleClick={handleClick} text="4" />
          <Button handleClick={handleClick} text="5" />
          <Button handleClick={handleClick} text="6" />
          <Button
            handleClick={handleClick}
            text="-"
            backgroundColor={orangeColor}
          />
        </div>

        <div style={rowStyles}>
          <Button handleClick={handleClick} text="1" />
          <Button handleClick={handleClick} text="2" />
          <Button handleClick={handleClick} text="3" />
          <Button
            handleClick={handleClick}
            text="+"
            backgroundColor={orangeColor}
          />
        </div>

        <div style={rowStyles}>
          <Button handleClick={handleClick} text="0" width="51%" />
          <Button handleClick={handleClick} text="." />
          <Button
            handleClick={handleClick}
            text="="
            backgroundColor={orangeColor}
          />
        </div>
      </div>
    </>
  );
};

export default Calculator;
