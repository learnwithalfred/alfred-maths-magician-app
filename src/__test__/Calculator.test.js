import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';
import '@testing-library/jest-dom';

describe('Calculator page should render', () => {
  it('test if Calculator page is rendered', () => {
    const calculatorTest = renderer.create(<Calculator />).toJSON();
    expect(calculatorTest).toMatchSnapshot();
  });
});
