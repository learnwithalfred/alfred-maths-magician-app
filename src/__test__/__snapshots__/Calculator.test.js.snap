// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page should render test if Calculator page is rendered 1`] = `
<div
  style={
    Object {
      "width": "500px",
    }
  }
>
  <div
    style={
      Object {
        "color": "black",
        "display": "flex",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <div
      className="input-element"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#858694",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "flex-end",
          "margin": "2px",
          "padding": "5px",
          "width": "100%",
        }
      }
    />
  </div>
  <div
    style={
      Object {
        "color": "black",
        "display": "flex",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5913E",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>
  <div
    style={
      Object {
        "color": "black",
        "display": "flex",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5913E",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      x
    </button>
  </div>
  <div
    style={
      Object {
        "color": "black",
        "display": "flex",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5913E",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>
  <div
    style={
      Object {
        "color": "black",
        "display": "flex",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5913E",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <div
    style={
      Object {
        "color": "black",
        "display": "flex",
        "flexWrap": "nowrap",
        "width": "100%",
      }
    }
  >
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "51%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#E0E0E0",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="undefined undefined"
      onClick={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#F5913E",
          "border": "none",
          "display": "flex",
          "fontSize": "1.3rem",
          "height": "50px",
          "justifyContent": "center",
          "margin": "2px",
          "outline": "none",
          "padding": "3px",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
